pkg/controller/replicaset/replica_set.go

```diff
@@ -255,9 +283,9 @@ func (rsc *ReplicaSetController) updateRS(old, cur interface{}) {
 
 // isCacheValid check if the cache is valid
 func (rsc *ReplicaSetController) isCacheValid(pod *api.Pod, cachedRS *extensions.ReplicaSet) bool {
-	_, err := rsc.rsLister.ReplicaSets(cachedRS.Namespace).Get(cachedRS.Name)
+	rs, err := rsc.rsLister.ReplicaSets(cachedRS.Namespace).Get(cachedRS.Name)
 	// rs has been deleted or updated, cache is invalid
-	if err != nil || !isReplicaSetMatch(pod, cachedRS) {
+	if err != nil || !isReplicaSetMatch(pod, cachedRS) || rs.ResourceVersion != cachedRS.ResourceVersion {
```

Comment:
Removed this check , it's redundant, if Spec.Selector has been changed the cach will be invalidated: https://github.com/kubernetes/kubernetes/blob/master/pkg/controller/replicaset/replica_set.go#L235